### DISCUSSION TITLE
Correct use of Scene Filename Parser

### DISCUSSION THREAD

**Discussion by LeGrosFromage** [stashapp/stash#2740] [Q&A] [answered]
Asking for what the correct method is to pull a Performer name file from a filename.

So what I have is some files which are structured like this:
Firstname lastname - title (date).ext

Actual example:
Chelsy Sun - Slot For A Slut (2013-12-16).mp4

Using the Scene Filename Parser, I've entered this recipe:
{performer}-{title}({date}).ext

The title and date are copied over correctly, but the performer name is not.  I don't know if it's because there's a space there, although I have done some with only a first name and nothing changed.  Is there a specific way to handle performer names, am I missing something?

**Comment by WithoutPants** [comment, marked as answer]
I was able to get it to parse correctly by using `{performer} - {title}({date}).{ext}`

Performer internally gets mapped to `.*` in the regex pattern, then stash tries to match the performer with the name. I think in this case the parser ends up with the trailing space, causing it to not match. We could probably change it to trim the leading and trailing whitespace to make it less finicky.

**Comment by LeGrosFromage** [comment]
Thanks for that. I hadn't been able to get it to work, in the meantime I'd simply done a copy/paste of the performer name into the Performer Name field. Took a while but I did it in phases so wasn't too much of a hardship. Thank you for the info though. I didn't know if the parser was seeing the space as a literal thing and needed escaping... I'm lousy with regex so the copy/paste was the easiest solution for me!
-------- Original Message --------
